File: Models/Notification.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using application_Livraison.Models;

public class Notification
{
    public int Id { get; set; }

    [Required]
    public string Message { get; set; }

    public DateTime SentAt { get; set; } = DateTime.Now;

    public bool IsRead { get; set; } = false;

    // Clés étrangères
    public int ExpediteurId { get; set; }

    [ForeignKey("ExpediteurId")]
    public User Expediteur { get; set; }

    public int DestinataireId { get; set; }

    [ForeignKey("DestinataireId")]
    public User Destinataire { get; set; }
}

File: src/app/models/notification.ts
// src/app/models/notification.ts

import { User } from './user';

export interface Notification {
  id: number;
  message: string;
  sentAt: Date;

  // Expéditeur (expediteurId et objet user complet facultatif selon l'API)
  expediteurId: number;
  expediteur?: User;

  // Chauffeur (destinataire)
  chauffeurId: number ;
  chauffeur?: User;
}
